pkg/kubectl/cmd/rollingupdate.go

```diff
@@ -120,10 +121,13 @@ func RunRollingUpdate(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, arg
 	}
 
 	var hasLabel bool
-	for key, oldValue := range oldRc.Spec.Selector {
-		if newValue, ok := newRc.Spec.Selector[key]; ok && newValue != oldValue {
-			hasLabel = true
-			break
+	for _, oldReq := range oldRc.Spec.Selector {
+		for _, newReq := range newRc.Spec.Selector {
+			if oldReq.Key == newReq.Key &&
+				!reflect.DeepEqual(oldReq.StrValues, newReq.StrValues) {
```

Comment:
I think the weird is th end line after the `&&`. I'll modify it. Anyway I think my editor ran `gofmt`...